packages/app-generic-asset/src/index.tsx

```diff
@@ -3,19 +3,41 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { AppProps, BareProps } from '@polkadot/react-components/types';
+import { useApi, useCall } from '@polkadot/react-hooks';
 
 import React, { useMemo } from 'react';
 import { Route, Switch } from 'react-router';
 import { Tabs } from '@polkadot/react-components';
 
 import Transfer from './Transfer';
 import Assets from './Assets';
+import assetsRegistry, {STAKING_ASSET_NAME, SPENDING_ASSET_NAME} from './assetsRegistry';
 
 import { useTranslation } from './translate';
+import { AssetId } from '@cennznet/types/runtime';
 
 interface Props extends AppProps, BareProps {}
 
 export default function AssetApp ({ basePath }: Props): React.ReactElement<Props> {
+  const { api } = useApi();
+
+  // Populate staking/CENNZ and spending/CPAY IDs from the connected chain
+  const stakingAssetId = useCall<AssetId>(api.query.genericAsset.stakingAssetId as any, []);
```

Comment:
queries more often than it should but it can be removed when we use GA asset metadata map